src/py/flwr/superexec/exec_servicer.py

```diff
@@ -138,15 +138,13 @@ def StopRun(
 
         # Exit if `run_id` not found
         if not state.get_run(request.run_id):
-            context.abort(
-                grpc.StatusCode.NOT_FOUND, f"Run ID {request.run_id} not found"
-            )
+            return StopRunResponse(success=False, reason="Run not found")
 
         run_status = state.get_run_status({request.run_id})[request.run_id]
+        # Exit if run already finished
         if run_status.status == Status.FINISHED:
-            context.abort(
-                grpc.StatusCode.FAILED_PRECONDITION,
-                f"Run ID {request.run_id} is already finished",
+            return StopRunResponse(
+                success=False, reason=f"Run ID {request.run_id} is already finished"
```

Comment:
I think the idea is to use the gRPC status codes and eventually unify the handling of the errors. There's also this earlier comment about using [statuscode](https://github.com/adap/flower/pull/4634#discussion_r1871099540).